tools/pip_install.py

```diff
@@ -99,8 +99,15 @@ def main(args):
         else:
             # Otherwise, we merge requirements to build the constraints and pin dependencies
             requirements = None
+            reinstall = False
```

Comment:
I don't think this PR should be making any changes to this file.

I think the reason for the merge conflicts here is cherry picking the changes from master to the point release branch changed the commits and we've modified `pip_install.py` since cherry picking those changes.